orbit/pkg/platform/platform_windows.go

```diff
@@ -176,3 +178,124 @@ func GetProcessByName(name string) (*gopsutil_process.Process, error) {
+	}
+
+	// Determine SMBIOS version and table location from entry point
+	biosMajor, biosMinor, _ := smBIOSRawData.Version()
+
+	// SMBIOS constants from spec here - https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.1.1.pdf
+	const systemInformationType uint8 = 0x01 // System Information indicator
+	const minBiosStructSize uint8 = 0x1b     // Section 7.2 on the SMBIOS specification (0x1a min header size + null character)
+	const uuidOffset uint8 = 0x4             // UUID offset in System Information (Type 1) structure
+	const revMajorVersion int = 0x3          // SMBIOS revision that most of the current BIOS have - v3 specs were released in 2015
+	const minLegacyMajorVersion int = 0x2    // Minimum SMBIOS Major rev that supports UUID little-endian encoding
+	const minLegacyMinorVersion int = 0x6    // Minimum SMBIOS Minor rev that supports UUID little-endian encoding
+	const uuidSize int = 0x10                // UUID size is calculated with field offset value (0xA) + node field length (6 bytes)
+
+	var smBiosUUID string = ""
+
+	// Walking the obtained SMBIOS data
+	for _, rawBiosStruct := range structSMBIOSdata {
+		if (rawBiosStruct.Header.Type == systemInformationType) && (rawBiosStruct.Header.Length >= minBiosStructSize) {
+			uuidBytes := rawBiosStruct.Formatted[uuidOffset:]
+
+			// UUID field sanity check for null values
+			// Logic is based on https://github.com/ContinuumLLC/godep-go-smbios/blob/ab7c733f1be8e55ed3e0587d1aa2d5883fe8801e/smbios/decoder.go#L135
+			only0xFF, only0x00 := true, true
+			for i := 0; i < uuidSize && (only0x00 || only0xFF); i++ {
+				if uuidBytes[i] != 0x00 {
+					only0x00 = false
+				}
+				if uuidBytes[i] != 0xFF {
+					only0xFF = false
+				}
+			}
+
+			if only0xFF {
+				return "", errors.New("UUID is not currently present in the system, but it can be set.")
+			}
+			if only0x00 {
+				return "", errors.New("UUID is not present in the system.")
+			}
+
+			// As off version 2.6 of the SMBIOS specification, the first 3 fields of the UUID are supposed to be encoded on little-endian(section 7.2.1)
+			if (biosMajor >= revMajorVersion) || (biosMajor >= minLegacyMajorVersion && biosMinor >= minLegacyMinorVersion) {
+				smBiosUUID = fmt.Sprintf("%02X%02X%02X%02X-%02X%02X-%02X%02X-%02X%02X-%02X%02X%02X%02X%02X%02X",
+					uuidBytes[3], uuidBytes[2], uuidBytes[1], uuidBytes[0], uuidBytes[5], uuidBytes[4], uuidBytes[7], uuidBytes[6], uuidBytes[8], uuidBytes[9], uuidBytes[10], uuidBytes[11], uuidBytes[12], uuidBytes[13], uuidBytes[14], uuidBytes[15])
+			} else {
+				smBiosUUID = fmt.Sprintf("%02X%02X%02X%02X-%02X%02X-%02X%02X-%02X%02X-%02X%02X%02X%02X%02X%02X",
+					uuidBytes[0], uuidBytes[1], uuidBytes[2], uuidBytes[3], uuidBytes[4], uuidBytes[5], uuidBytes[6], uuidBytes[7], uuidBytes[8], uuidBytes[9], uuidBytes[10], uuidBytes[11], uuidBytes[12], uuidBytes[13], uuidBytes[14], uuidBytes[15])
+			}
+
+			break
+		}
+	}
+
+	return smBiosUUID, nil
+}
```

Comment:
Could be nice to refactor the parsing into another function and add a couple of unit tests since this seems like fairly complex logic.

Reply:
Good point. I've extracted the sanity check on the UUID field and added some UT to it